/*
 *  Copyright (c) 2020 NetEase Inc.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

syntax = "proto2";
package curve.chunkserver;

option cc_generic_services = true;

message ScanMap {
    required uint32 logicPoolId = 1;
    required uint32 copysetId = 2;
    required uint64 chunkId = 3;
    required uint64 index = 4;
    required uint32 crc = 5;
    required uint64 offset = 6;
    required uint64 len = 7;
};

message FollowScanMapRequest {
    required ScanMap scanMap = 1;
}

message FollowScanMapResponse {
    required sint32 retCode = 1;    
}

service ScanService {
    rpc FollowScanMap(FollowScanMapRequest) returns (FollowScanMapResponse);
}
